### PR TITLE
Add WalletPicker component for connecting to multiple blockchain networks and wallets

### PR DESCRIPTION
**Pull request by leirbag95**
This pull request adds a new component, WalletPicker, to allow users to easily connect their wallets to various blockchain networks and choose their preferred wallet provider.

The **WalletPicker** component provides a user-friendly interface with networks menus to select the blockchain network and then wallet provider. Users can select from a range of popular wallet providers such as Metamask, Phantom, or even Polka.js. 

This component is designed to be highly customizable, and can be easily integrated into existing applications. It provides a simple and streamlined way for users to connect to their preferred wallets on their preferred blockchain networks.

Capture d’écran 2023-03-28 à 12 08 10

### CODE REVIEW

File: src/components/WalletPicker/cmp.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import Icon from '../Icon'
+import { StyledPicker, Text } from './styles'
+import { PickerProps } from './types'
+
+export const WalletPicker = forwardRef(({
+  variant = 'primary',
+  kind = 'flat',
+  size = 'regular',
+  color = 'main2',
+  networks = [],
```
Comment (BjrInt) [inline src/components/WalletPicker/cmp.tsx:13]: These are not defined in the `types.ts` file. Can you :
- Create a type definition for the `network` object
- Remove unused props (ex: color, kind, variant)

File: src/components/WalletPicker/cmp.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    name: 'WalletConnect',
+    icon: 'circle',
+    color: 'cyan',
+    callback: null
+  },
+  phantom: {
+    name: 'Phantom',
+    icon: 'circle',
+    color: 'purple',
+    callback: null
```
Comment (BjrInt) [inline src/components/WalletPicker/cmp.stories.tsx:36]: If callback is supposed to be a function provide a noop `() => null` as a default value instead (so you don't run into a `TypeError`)

File: src/components/WalletPicker/types.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+export type PickerSize = 'regular'
+
+export type PickerProps =  {
+  children: React.ReactNode
+  size: PickerSize
+  color: string
+
+  // Force states
+  className?: string
```
Comment (BjrInt) [inline src/components/WalletPicker/types.ts:10]: Why is there a className property?

**Review by BjrInt — CHANGES_REQUESTED** [review]
Can you also provide a working example with an actual provider (ex: instanciating Metamask once the button is clicked?)

File: src/components/index.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ export { default as Table } from './Table'
 export { default as Tabs } from './Tabs'
 export { default as Tag } from './Tag'
 export { default as TextGradient } from './TextGradient'
+export { default as WalletPicker } from './WalletPicker'
```
[the remark this answers is not in the document]
Comment (leirbag95) [reply src/components/index.ts:14]: Done, I pull and rebase main branch into Feat/WalletPicker branch

File: src/components/WalletPicker/cmp.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        color,
+        ...rest
+      }}
+    >
+      <div>
+        <Text>1. Choose your network</Text>
+        <Row count={4}>
+          {
+            (
+              networks.map((network) => (
```
[the remark this answers is not in the document]
Comment (leirbag95) [reply src/components/WalletPicker/cmp.tsx:38]: Now, NetworkProps has been created 👍

File: src/components/WalletPicker/types.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+export type PickerSize = 'regular'
+
+export type PickerProps =  {
+  children: React.ReactNode
+  size: PickerSize
```
[the remark this answers is not in the document]
Comment (leirbag95) [reply src/components/WalletPicker/types.ts:6]: I added another kind of size : 'large' with bigger text, width and icon

File: src/components/WalletPicker/cmp.stories.tsx (hunk 2 above)
Comment (leirbag95) [reply src/components/WalletPicker/cmp.stories.tsx]: This example does not exist anymore

File: src/components/WalletPicker/cmp.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+
+const networks = [
+  { name: 'Ethereum',  icon: 'ethereum', wallets: [wallets.metamask, wallets.wc] },
+  { name: 'Bitcoin',   icon: 'bitcoin',  wallets: [] }
+]
+
+const defaultArgs: Partial<PickerProps> = {
+  color: 'main2',
```
[the remark this answers is not in the document]
Comment (leirbag95) [reply src/components/WalletPicker/cmp.stories.tsx]: replaced it by backgroundColor

File: src/components/WalletPicker/cmp.tsx (hunk 1 above)
Comment (leirbag95) [reply src/components/WalletPicker/cmp.tsx]: NetworkProps and WalletProps has been created

**Pull request closed without merging**